### PR TITLE
[2.5] Update KDM with updated hyperkube-base

### PR DESCRIPTION
**Pull request by rayandas**
Related: https://github.com/rancher/rancher/issues/36021

### CODE REVIEW

File: rke/k8s_rke_system_images.go [hunk 1]
```diff
@@ -6085,6 +6085,50 @@ func loadK8sRKESystemImages() map[string]v3.RKESystemImages {
 			WindowsPodInfraContainer:  "rancher/kubelet-pause:v0.1.6",
 			Nodelocal:                 "rancher/mirrored-k8s-dns-node-cache:1.15.13",
 		},
+		// Enabled in v2.5.12
+		"v1.20.14-rancher2-2": {
```
Comment (snasovich) [inline rke/k8s_rke_system_images.go]: I think this should be `v1.20.14-rancher2-1` as we don't have `v1.20.14-rancher2-1` yet. @kinarashah , please correct me if I'm wrong here.

**Review by snasovich — CHANGES_REQUESTED** [review]

File: rke/k8s_rke_system_images.go (hunk 1 above)
Comment (rayandas) [reply rke/k8s_rke_system_images.go]: `v1.20.14-rancher1-2` was there and it was merged 5 days ago so I just copied and used that template.

Comment (kinarashah) [reply rke/k8s_rke_system_images.go]: @snasovich Yeah, that's correct. In `rancherX-Y`, `X` is for hyperkube bumps and `Y` is for bumping system images or addon templates. Here we are updating hyperkube, so we'd use rancher2-1. 

@rayandas Also, we already have a new k8s version for v1.20.14 in KDM, v1.20.14-rancher1-2 - it was introduced for [metrics server bump](https://github.com/rancher/kontainer-driver-metadata/pull/783/files). This version hasn't been released yet, so we can update it to v1.20.14-rancher2-1 (and keep updating it till it's merged to release-v2.5 branch).

Comment (rayandas) [reply rke/k8s_rke_system_images.go]: ah! alright. I got it. Updating both the PRs. Thanks.

**Review by snasovich — APPROVED** [review]

**Pull request merged**